### PR TITLE
[JSC] Align JIT and LLInt registers in x64

### PR DESCRIPTION
**Pull request by Constellation**
#### ed54aa241721571d1de5d06856825d2e4abeb32e
<pre>
[JSC] Align JIT and LLInt registers in x64
<a href="https://bugs.webkit.org/show_bug.cgi?id=279642">https://bugs.webkit.org/show_bug.cgi?id=279642</a>
<a href="https://rdar.apple.com/135933164">rdar://135933164</a>

Reviewed by Ross Kirsling.

Let&apos;s align both. There is no reason having different namings anymore (no Windows).

* Source/JavaScriptCore/jit/GPRInfo.h:
* Source/JavaScriptCore/llint/LowLevelInterpreter.asm:
* Source/JavaScriptCore/llint/LowLevelInterpreter64.asm:
* Source/JavaScriptCore/llint/WebAssembly64.asm:
* Source/JavaScriptCore/offlineasm/x86.rb:

Canonical link: <a href="https://commits.webkit.org/283620@main">https://commits.webkit.org/283620@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/0c09b44a760cacd8efe88ae193c9115f437bb3e9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66821 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46196 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19443 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70856 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/17954 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53995 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17730 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53522 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12084 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69888 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42503 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57811 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34154 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39175 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15203 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16308 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/59935 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61087 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15544 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72557 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/66066 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10778 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14897 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60933 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10810 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57868 "Passed tests") | [loading-orange 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61167 "Build is in progress. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; 2 api tests failed or timed out; 2 api tests failed or timed out; Compiled WebKit; 2 api tests failed or timed out; Running analyze-api-tests-results") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8849 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2472 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/87834 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10139 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42003 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/15450 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43080 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44263 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42823 "Built successfully") | | |

### CODE REVIEW

File: Source/JavaScriptCore/offlineasm/x86.rb [hunk 1]
```diff
@@ -177,66 +177,43 @@ def supports8BitOnX86
     end
 
     def x86GPR
-        if isX64
```
Comment (rkirsling) [inline Source/JavaScriptCore/offlineasm/x86.rb:180]: Hmm, seems like `isX64` should be removed altogether if it just throws in the false case now...?

File: Source/JavaScriptCore/offlineasm/x86.rb [hunk 2]
```diff
@@ -29,13 +29,13 @@
 # On x86-64 (windows and non-windows)
 #
 # rax => t0,     r0
-# rdi =>     a0
+# rdi => t6  a0
```
Comment (rkirsling) [inline Source/JavaScriptCore/offlineasm/x86.rb]: tiniest nit ever
```suggestion
# rdi => t6, a0
```

**Review by rkirsling — APPROVED** [review]
Cool!

File: Source/JavaScriptCore/offlineasm/x86.rb (hunk 2 above)
Comment (Constellation) [reply Source/JavaScriptCore/offlineasm/x86.rb]: Fixed.

File: Source/JavaScriptCore/offlineasm/x86.rb (hunk 1 above)
Comment (Constellation) [reply Source/JavaScriptCore/offlineasm/x86.rb:180]: Sounds good.

Comment (MenloDorian) [reply Source/JavaScriptCore/offlineasm/x86.rb:180]: Maybe the file should be renamed from x86.rb to x86_64.rb (or x64.rb) to make it clear that it does not support 32-bit x86 anymore.  I suggest doing that in a subsequent patch by itself.

Comment (Constellation) [reply Source/JavaScriptCore/offlineasm/x86.rb:180]: Sounds good too!

**Pull request merged**